### PR TITLE
add batchspawner 1.1.0

### PR DESCRIPTION
**Pull request by bollwyvl**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

File: recipes/batchspawner/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  script: {{ PYTHON }} -m pip install . -vv --no-deps
+  number: 0
+
+requirements:
+  host:
+    - __unix
+    - pip
+    - python >=3.6
+  run:
+    - __unix
```
Comment (bollwyvl) [inline recipes/batchspawner/meta.yaml:22]: upstream only claims macos/linux... could remove this though

**Comment by bollwyvl** [comment]
@conda-forge/staged-recipes @conda-forge/help-python ready for review! a windows fail is expected.

File: recipes/batchspawner/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  name: batchspawner
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/b/batchspawner/batchspawner-{{ version }}.tar.gz
+  sha256: 9bae72f7c1bd9bb11aa58ecc3bc9fae5475a10fdd92dc0c0d67fa7eb95c9dd3a
+
+build:
+  noarch: python
+  script: {{ PYTHON }} -m pip install . -vv --no-deps
```
Comment (BastianZim) [inline recipes/batchspawner/meta.yaml:13]: ` --no-deps` should be taken care off by `{{ PYTHON }}` or do you explicitly need it?
```suggestion
  script: {{ PYTHON }} -m pip install . -vv
```

File: recipes/batchspawner/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  number: 0
+
+requirements:
+  host:
+    # only targets unix systems, imports `pwd`
+    - __unix
```
Comment (BastianZim) [inline recipes/batchspawner/meta.yaml:19]:
```suggestion
```

This is only required for run. Doesn't hurt here, just for "cleanliness".

File: recipes/batchspawner/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    - pip
+    - python >=3.6
```
Comment (BastianZim) [inline recipes/batchspawner/meta.yaml:21]: Upstream has `>=3.3`. Up to you which one to use.

Ref: https://github.com/jupyterhub/batchspawner/blob/f4659678061dacf5837fb15ce340230dca18f3ca/setup.py#L43

File: recipes/batchspawner/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    - jinja2
+    - jupyterhub-base >=0.5
+    - python >=3.6
+
+test:
+  imports:
+    - batchspawner
+  requires:
+    - pip
+    - packaging  # TODO: remove after https://github.com/conda-forge/jupyterhub-feedstock/issues/59
```
Comment (BastianZim) [inline recipes/batchspawner/meta.yaml:34]: Shouldn't this be under run? Seems like it's needed for the package.

**Review by BastianZim — COMMENTED** [review]
Just a couple of comments. Just let me know if you want to change them, otherwise I'll merge it.

File: recipes/batchspawner/meta.yaml (hunk 5 above)
Comment (bollwyvl) [reply recipes/batchspawner/meta.yaml:32]: This one doesn't import `packaging`, and indeed, is only required by jupyterhub 2+

https://github.com/jupyterhub/jupyterhub/pull/3562

I've got a PR for that feedstock to fix:

https://github.com/conda-forge/jupyterhub-feedstock/pull/60

File: recipes/batchspawner/meta.yaml (hunk 4 above)
Comment (bollwyvl) [reply recipes/batchspawner/meta.yaml]: fair!

**Comment by bollwyvl** [comment]
@BastianZim thanks for the speedy review! i think if it comes back up :heavy_check_mark:  :heavy_check_mark:  :exclamation: again, we're good!

File: recipes/batchspawner/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+  commands:
+    - pip check
+    # even this expects a running hub
+    # - batchspawner-singleuser --version
```
Comment (jakirkham) [inline recipes/batchspawner/meta.yaml:36]: Maybe could do `command -v batchspawner-singleuser` or check for this in `$PREFIX/bin` along with something similar on Windows

Comment (bollwyvl) [reply recipes/batchspawner/meta.yaml:36]: > something similar on Windows

_luckily_ not a concern. I intend to try to get a `--version` added upstream, and use `console_scripts`.

**Pull request merged**